### DISCUSSION TITLE
Are you aware of timing issue with first iteration(s) of Midifile.play()?

### DISCUSSION THREAD

**Discussion by Frnot** [mido/mido#469] [Q&A] [answered]
I have been having an issue using mido for .mid playback for almost a year now. I finally decided to hunt down the source, but I'm curious if it's a known issue.

The MidiFile.__iter__() function calls merge_tracks() on its first iteration. I have examples of this function taking many tenths of seconds to execute. This causes issues when using Midifile.play() to provide the timing for midi playback, as the first note(s) will play long after their calculated 'input_time' (because they have to wait on merge_tracks()). As you can imagine, the start of playback sounds nothing like it should when playing these files using the built in play() generator.

From what I can tell, the fix is trivial. Add a merged_track instance variable to MidiFile and call merge_tracks() at the end of MidiFile._load() and MidiFile.add_track().

I could only find one reference to this issue (nearly six years ago) but no proposed solution. I'd be happy to submit a pull request if this issue is deemed relevant and the proposed solution is acceptable.

**Comment by almostimplemented** [comment]
I saw your PR and suggested a different solution: to use a min-heap to sort the track iterators instead of the up-front computation. 

But I agree it should be fixed one way or another! :+1:

**Comment by rdoursenaud** [comment, marked as answer]
I wasn't. Thanks for your comments, analysis (#472) and pull requests (#470, #474)!
I'll have a better look as soon as I'm able to free up some time ;)
